### DISCUSSION TITLE
Listen on Unix Sockets

### DISCUSSION THREAD

**Discussion by shtrophic** [dani-garcia/vaultwarden#3888] [Q&A]
Is it possible to make vaultwarden listen on an unix socket instead of the network?
I am using a reverse proxy that handles the network connections. I think it would be preferable to tell the reverse proxy to forward it to a socket, since then you could leverage unix filesystem permissions to restrict other programs sending requests to vaultwarden, for additional defense in depth.

**Comment by BlackDex** [comment]
Not possible: https://github.com/SergioBenitez/Rocket/issues/1070

**Reply by shtrophic** [reply]
soon™

**Reply by septatrix** [reply]
This is now supported by rocket

**Reply by BlackDex** [reply]
In master and not v0.5 which is the current stable version, and Vaultwarden uses.
So do not think this will be implemented any time soon :)

**Reply by polyzen** [reply]
Just to note: 0.5.1 was just released, which appears to not have this feature included yet.

**Reply by BlackDex** [reply]
I think it will be part of 0.6 actually.
I created a PR to at least update rocket :)

**Reply by VityaSchel** [reply]
are there any plans to implement this? :)

**Reply by BlackDex** [reply]
As long as there is no new release of Rocket this will not happen.

**Reply by VityaSchel** [reply]
> As long as there is no new release of Rocket this will not happen.

Wasn't this implemented like 3 years ago?

**Reply by BlackDex** [reply]
In a non-stable release probably, But there has not been a new release of Rocket since May 2024, and the v0.5 release does not have this feature.

**Reply by unoukujou** [reply]
> In a non-stable release probably, But there has not been a new release of Rocket since May 2024, and the v0.5 release does not have this feature.

@BlackDex maybe this can help:

https://crates.io/crates/rkt
https://github.com/rustfoo/rkt

I don't think Rocket is actively developed and Unix socket is really a good security and performance upgrade to have as an option.
